src/main/java/duke/logic/Parser.java

```diff
@@ -0,0 +1,98 @@
+package duke.logic;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.EventCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.InvalidCommand;
+import duke.command.ListCommand;
+import duke.command.TodoCommand;
+import duke.exception.DukeException;
+
+/**
+ * Represents a helper class that deals with making sense of the user command.
+ */
+public class Parser {
+
+    /**
+     * Returns a command instance after parsing the input command.
+     *
+     * @param fullCommand The command to be parsed.
+     * @return A command instance which its type is determined by the input command.
+     * @throws DukeException If the input command is deemed invalid or the format is incorrect.
+     */
+    public static Command parseCommand(String fullCommand) throws DukeException {
+        Command command;
+        String[] splitCommand = fullCommand.split(" ", 2);
+        String action = splitCommand[0];
+        String errorMessage = "";
+        try {
+            if (action.equals("bye")) {
```

Comment:
Perhaps you can consider using a switch-case block instead of multiple if-else loops.  That will make the code more readable
switch(action) {
case("bye"):
    do something
}